hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -2808,6 +2808,35 @@ def reverse(self):
         """
         return self._method('reverse', tstr)
 
+    @typecheck_method(collection=expr_oneof(expr_array(), expr_set()))
+    def join(self, collection):
+        """Returns a string which is the concatenation of the strings in `collection`
+        separated by the string providing this method. Raises :class:`TypeError` if
+        the element type of `collection` is not :data:`.tstr`.
+
+        Examples
+        --------
+
+        >>> a = ['Bob', 'Charlie', 'Alice', 'Bob', 'Bob']
+
+        >>> hl.eval(hl.str(',').join(a))
+        'Bob,Charlie,Alice,Bob,Bob'
+
+        Parameters
+        ----------
+        collection : :class:`.ArrayExpression` or :class:`.SetExpression`
+            Collection.
+
+        Returns
+        -------
+        :class:`.StringExpression`
+            Joined string expression.
+        """
+        if collection.dtype.element_type != tstr:
+            raise TypeError(f"Expected str collection, {collection.dtype.element_type} found")
+
+        return collection._method("mkString", tstr, self)
```

Comment:
prefer to just call `hl.delimit`, since then we can change the implementation in once place.